pkg/controllers/controller.go

```diff
@@ -46,14 +46,17 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	// 2. Copy object for merge patch base
 	persisted := resource.DeepCopyObject()
 	// 3. Reconcile
-	if err := c.Controller.Reconcile(ctx, resource); err != nil {
-		resource.StatusConditions().MarkFalse(v1alpha1.Active, "", err.Error())
-		zap.S().Errorf("Controller failed to reconcile kind %s, %s",
-			resource.GetObjectKind().GroupVersionKind().Kind, err.Error())
+	var reconcileError error
+	if reconcileError = c.Controller.Reconcile(ctx, resource); reconcileError != nil {
+		resource.StatusConditions().MarkFalse(v1alpha1.Active, "", reconcileError.Error())
+		reconcileError = fmt.Errorf("Controller failed to reconcile kind %s, %s",
+			resource.GetObjectKind().GroupVersionKind().Kind, reconcileError.Error())
+		zap.S().Errorf(reconcileError.Error())
+		return reconcile.Result{Requeue: true}, nil
```

Comment:
Oh this is kind of brilliant and way more elegant than returning an error. Nice!